pkg/certs/generate.go

```diff
@@ -0,0 +1,115 @@
+package certs
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"time"
+)
+
+// NewCACertificate generates and signs new CA certificate and key.
+func NewCACertificate(subject pkix.Name, serialNumber int64, currentTime func() time.Time) (*Certificate, error) {
+	caPublicKey, caPrivateKey, err := newKeyPair()
+	if err != nil {
+		return nil, err
+	}
+
+	var caLifetimeInDays = CABundleValidForDays
```

Comment:
why the var?

Reply:
Replaced with construction such as `lifetime := time.Duration(ClientBundleValidForDays) * 24 * time.Hour`.